source/dialogs/window.c

```diff
@@ -369,9 +397,46 @@ static int window_match ( const Mode *sw, rofi_int_matcher **tokens, unsigned in
         }
     }
 
+    TICK_N ( "end window_match" );
     return match;
 }
 
+static void window_mode_parse_fields ()
```

Comment:
this function can be executed twice is both window and windowcd are enabled. It might be good to add a guard.

Reply:
Isn't it the goal of the line `if ( mode_get_private_data ( sw ) == NULL )` to ensure that data aren't initialised twice ?
```c
static int window_mode_init ( Mode *sw )
{
    if ( mode_get_private_data ( sw ) == NULL ) { 
        ModeModePrivateData *pd = g_malloc0 ( sizeof ( *pd ) );
        pd->window_regex = g_regex_new ( "{[-\\w]+(:-?[0-9]+)?}", 0, 0, NULL );
        mode_set_private_data ( sw, (void *) pd );
        _window_mode_load_data ( sw, FALSE );
        window_mode_parse_fields (); 
    }   
    return TRUE;
}
static int window_mode_init_cd ( Mode *sw )
{
    if ( mode_get_private_data ( sw ) == NULL ) { 
        ModeModePrivateData *pd = g_malloc0 ( sizeof ( *pd ) );
        pd->window_regex = g_regex_new ( "{[-\\w]+(:-?[0-9]+)?}", 0, 0, NULL );
        mode_set_private_data ( sw, (void *) pd );
        _window_mode_load_data ( sw, TRUE );
        window_mode_parse_fields (); 
    }   
    return TRUE;
}
```

Reply:
Yep, but different modi, with each it own private data.

Reply:
Oh, ok, thanks !